pythonforandroid/toolchain.py

```diff
@@ -582,6 +586,14 @@ def add_parser(subparsers, *args, **kwargs):
             args.unknown_args += ["--private", args.private]
         if hasattr(args, "build_mode") and args.build_mode == "release":
             args.unknown_args += ["--release"]
+        if hasattr(args, "assets") and args.assets is not None:
+            # Pass this value on to the internal bootstrap build.py:
+            for asset in args.assets:
+                if ":" in asset:
+                    asset_src, asset_dest = asset.split(":")
+                else:
+                    asset_src = asset_dest = asset
+                args.unknown_args += ["--asset", os.path.abspath(asset_src)+":"+asset_dest]
```

Comment:
Isn't this just unnecessary? The unknown_args include everything that wasn't parsed by the first p4a cmdparser, so if you just don't have a `--add-asset` in that argumentparser it will get passed through anyway. It seems you're pulling out the arguments, then adding them again (just with --asset instead of --add-asset)?

Reply:
AFAIK, the build.py is executed in a different working directory, so we need to convert the relative paths to absolute before all that. The unknown_args part is not the problem, `os.path.abapath` is.

Reply:
Ah, got it, that's fine then, we can leave that code pattern for fixing later. Thanks for the clarification.